azurerm/resource_arm_api_management.go

```diff
@@ -0,0 +1,773 @@
+		certificate := config["certificate"].(string)
+		certificate_password := config["certificate_password"].(string)
+		default_ssl_binding := config["default_ssl_binding"].(bool)
+		negotiate_client_certificate := config["negotiate_client_certificate"].(bool)
+
+		hostname := apimanagement.HostnameConfiguration{
+			Type:                       host_type,
+			HostName:                   &host_name,
+			EncodedCertificate:         &certificate,
+			CertificatePassword:        &certificate_password,
+			DefaultSslBinding:          &default_ssl_binding,
+			NegotiateClientCertificate: &negotiate_client_certificate,
+		}
+
+		hostnames = append(hostnames, hostname)
+	}
+
+	return &hostnames
+}
+
+func expandAzureRmApiManagementCertificates(d *schema.ResourceData) *[]apimanagement.CertificateConfiguration {
+	certConfigs := d.Get("certificate").([]interface{})
+
+	certificates := make([]apimanagement.CertificateConfiguration, 0)
+
+	for _, v := range certConfigs {
+		config := v.(map[string]interface{})
+
+		cert_base64 := config["encoded_certificate"].(string)
+		certificate_password := config["certificate_password"].(string)
+		store_name := apimanagement.StoreName(config["store_name"].(string))
+
+		cert := apimanagement.CertificateConfiguration{
+			EncodedCertificate:  &cert_base64,
+			CertificatePassword: &certificate_password,
+			StoreName:           store_name,
+		}
+
+		certificates = append(certificates, cert)
+	}
+
+	return &certificates
+}
+
+func expandAzureRmApiManagementAdditionalLocations(d *schema.ResourceData, sku *apimanagement.ServiceSkuProperties) *[]apimanagement.AdditionalLocation {
+	inputLocations := d.Get("additional_location").([]interface{})
+
+	additionalLocations := make([]apimanagement.AdditionalLocation, 0)
+
+	for _, v := range inputLocations {
+		config := v.(map[string]interface{})
+		location := config["location"].(string)
+
+		additionalLocation := apimanagement.AdditionalLocation{
+			Location: &location,
+			Sku:      sku,
+		}
+
+		additionalLocations = append(additionalLocations, additionalLocation)
+	}
+
+	return &additionalLocations
+}
+
+func expandAzureRmApiManagementSku(configs []interface{}) *apimanagement.ServiceSkuProperties {
+	config := configs[0].(map[string]interface{})
+
+	nameConfig := config["name"].(string)
+	name := apimanagement.SkuType(nameConfig)
+	capacity := int32(config["capacity"].(int))
+
+	sku := &apimanagement.ServiceSkuProperties{
+		Name:     name,
+		Capacity: &capacity,
+	}
+
+	return sku
+}
+
+func flattenApiManagementCertificates(d *schema.ResourceData, props *[]apimanagement.CertificateConfiguration) []interface{} {
+	certificates := make([]interface{}, 0)
+
+	if props != nil {
+		for i, prop := range *props {
+			certificate := make(map[string]interface{}, 0)
+
+			certificate["store_name"] = string(prop.StoreName)
+
+			// certificate password isn't returned, so let's look it up
+			passwKey := fmt.Sprintf("certificate.%d.certificate_password", i)
```

Comment:
You will have to look these up by name, if someone edits this on the portal the index's may no longer match.